### PR TITLE
Hips2fits new service

### PR DESCRIPTION
**Pull request by bmatthieu3**
Hi @bsipocz @adamginsburg @tboch

I have implemented a new package responsible for querying hips2fits: http://alasky.u-strasbg.fr/hips-image-services/hips2fits
This service take a HiPS name + astropy WCS and return a FITS/JPG/PNG image cutout of the HiPS for that WCS. The list of valid hips name is here: http://aladin.unistra.fr/hips/list
There are two methods: 
* **query_with_wcs** that accepts a HiPS name + an astropy.wcs
* **query** doing the same but accepts different parameters instead of an astropy wcs (the center of projection, fov angle, projection, rotation...)

I added a simple doc containing 2 examples with the corresponding output plot images that can be obtained thanks to hips2fits. Some tests (remote too) have also been added.

Thank you very much for your reviews.

### CODE REVIEW

File: docs/hips2fits/hips2fits.rst [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    See `here <http://aladin.unistra.fr/hips/list>`_ all the valid HiPS names hosted in CDS.
+* :meth:`~astroquery.hips2fits.hips2fitsClass.query_without_wcs` extracting a FITS image from a HiPS given the output image pixel size, the center of projection, the type of projection and the field of view.
+    See `here <http://aladin.unistra.fr/hips/list>`_ all the valid HiPS names hosted in CDS.
+
+
+Examples
+========
+
+With an user defined astropy WCS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (keflavich) [inline docs/hips2fits/hips2fits.rst:35]:
```suggestion
With a user defined astropy WCS
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: astroquery/hips2fits/core.py [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+        # If a wcs is still present in the kwargs it means
+        # the user called query_with_wcs
+        if kwargs.get('wcs'):
+            wcs = kwargs.pop('wcs')
+            header = wcs.to_header()
+
+            # The WCS header does not contain the NAXISx keywords
+            # We can get them from the WCS directly and add them
+            # to the header
+            nx, ny = wcs.pixel_shape
```
Comment (keflavich) [inline astroquery/hips2fits/core.py:342]: you may not be able to rely on this being present; if a wcs was created not from a fits Header (i.e., from scratch), it won't have this attribute populated

Comment (bmatthieu3) [reply astroquery/hips2fits/core.py]: The hips2fits service asks for the size of the result image (width and height), it is mandatory. I changed the code, if the wcs has no pixel shape (pixel_shape  None) then I raise an AttributeError saying either to give your wcs the pixel size of the image or to use the other **query** method

Comment (bmatthieu3) [reply astroquery/hips2fits/core.py]: 50f14de

**Comment by bmatthieu3** [comment]
Hi @adamginsburg,

I have seen a test is failing because it runs tests from the doc api (https://travis-ci.org/github/astropy/astroquery/jobs/695927179) which are remote tests. How to say these doc tests are remote and thus can be skipped ? Thanks

PS: I also added a changelog

**Comment by keflavich** [comment]
Add `__doctest_skip__ = ['Hips2FitsClass.*']` at the top of the doc

**Comment by keflavich** [comment]
I don't have permission to push to this branch.  It needs a simple rebase to handle changelog conflicts.

**Comment by bsipocz** [comment]
@keflavich - Does this need another round of review, or mergeable pending resolving the conflict and getting an all green CI?

**Comment by bmatthieu3** [comment]
Hi @keflavich @bsipocz,

I added ```__doctest_skip__ = ['Hips2FitsClass.*']``` to the beginning of the core.py file so that doc tests do not execute (because they are remote).
I do not see the CI tests running.
We are very excited to merge this into the next version of astroquery. Let me know what I should do to make this merged.

**Comment by keflavich** [comment]
I'm going to merge this on the command line - we need to put the changelog entry into the right section, but otherwise it's good to go.

**Comment by keflavich** [comment]
The remote tests pop up a blocking `imshow` - we should avoid that!  However, the tests all passed.

File: astroquery/hips2fits/core.py [hunk 3]
```diff
@@ -10,6 +10,7 @@
 from astropy import wcs
 
 __all__ = ['hips2fits', 'hips2fitsClass']
+__doctest_skip__ = ['hips2fitsClass.*']
```
Comment (bsipocz) [inline astroquery/hips2fits/core.py:13]: @tinumide - when we get here, we should also look for doctest skips like this in the code files, and turn on the remote testing for them.

**Comment by bsipocz** [comment]
@bmatthieu3 @keflavich - Please open a follow-up PR for fixing the imshow issue or skip the test, as we want to turn back on the remote CI testing soonish.

File: astroquery/hips2fits/core.py (hunk 3 above)
Comment (tinumide) [reply astroquery/hips2fits/core.py:13]: Alright, noted

**Comment by bsipocz** [comment]
Retrospect I think this should have been a new module under cds, as part of our upcoming restructure.

**Pull request merged**